subiquity/cloudinit.py

```diff
@@ -67,6 +86,58 @@ def read_legacy_status(stream):
     return None
 
 
+def read_recoverable_errors(
+    stream: bytes | str,
+    level: Optional[str] = None,
+) -> list[str]:
+    """Retrieve recoverable errors from cloud-init status."""
+
+    try:
+        status: dict[str, Any] = json.loads(stream)
+    except json.JSONDecodeError:
+        log.warning("Failed to decode cloud-init status output")
+        return []
+
+    errors_by_level: dict[str, list[str]] = status.get("recoverable_errors", {})
+
+    if level is not None:
+        return errors_by_level.get(level, [])
+
+    all_errors: list[str] = []
+    for _level in errors_by_level:
+        all_errors.extend(errors_by_level[_level])
+
+    return all_errors
+
+
+async def get_schema_failure_sources() -> list[str]:
+    """Retrieve the keys causing schema failure."""
+
+    cmd: list[str] = ["cloud-init", "schema", "--system"]
```

Comment:
I think in general we want to capture errors from all sources since we read the combined config when extracting the autoinstall config (e.g. is it possible someone sends autoinstall config in their network-config?), but capturing the file source would improve our error messaging for sure. What do you think about leaving this to future improvements?